### DISCUSSION TITLE
Using Events on Graphic, How to identify which item was clicked?

### DISCUSSION THREAD

**Discussion by sketchpunk** [pixijs/pixijs#7774] [Q&A]
I see that Graphic allows to set itself as interactive and in buttonMode. The click event triggers and when I put the mouse over a drawn graphic the cursor turns to a pointer which is great. The one thing I can't figure out is how to identity what's been clicked? When i draw into graphics is there a way to identify each item that can be viewed through events?

Here's a playground with it working so far.  Will need to open dev tools to see the console.log when you click on an item.
https://www.pixiplayground.com/#/edit/o_7oqb6aHWNq_p-uN6Xx-

**Comment by ivanpopelyshev** [comment]
Not possible without adding your methods to the Graphics. You have to do custom containsPoint() .
Here's how current works:
https://github.com/pixijs/pixijs/blob/7ee2be5cf88bb990166db10cfb017e37aa2cbbb1/packages/graphics/src/Graphics.ts#L1109-L1114

https://github.com/pixijs/pixijs/blob/7ee2be5cf88bb990166db10cfb017e37aa2cbbb1/packages/graphics/src/GraphicsGeometry.ts#L297-L349

you can do it yourself or put a bounty on the task.

**Reply by sketchpunk** [reply]
If I was able to add functionality to graphic to be able identity an item. Is it something worth creating a PR for the project? Any code styles / rules I need to follow for any code submissions? Thanks for giving me a starting place to look.

**Reply by ivanpopelyshev** [reply]
at least an example  that we'll put to guides  :) No, you can trry PR https://github.com/pixijs/guides or just leave example here and some guys will move it to guides

**Reply by ivanpopelyshev** [reply]
use hacks like `Graphics.prototype.xxx = myMethod`  in example and that'll be enough
